staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -976,30 +976,37 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 			}
 		}
 
-		if celContext.HasValidationRules() && celContext.TypeInfo() != nil {
-			compResults, err := cel.Compile(celContext.TypeInfo().Schema, celContext.TypeInfo().DeclType, cel.PerCallLimit)
+		if celContext != nil {
+			typeInfo, err := celContext.TypeInfo()
 			if err != nil {
-				allErrs = append(allErrs, field.InternalError(fldPath.Child("x-kubernetes-validations"), err))
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("x-kubernetes-validations"), nil, fmt.Sprintf("internal error: failed to construct type information for x-kubernetes-validations rules: %s", err)))
```

Comment:
if we're calling this "internal error:" should we use `field.InternalError`? a `field.Invalid` with a `nil` value makes for a confusing error message (same question on the next error)

Reply:
My mistake, using `InternalError` everywhere now.